js_modules/dagster-ui/packages/ui-core/src/instance/InstanceConcurrencyKeyInfo.tsx

```diff
@@ -0,0 +1,641 @@
+            </Box>
+            <Box padding={{vertical: 16, horizontal: 24}}>
+              <Subheading>{flagPoolUI ? 'Pool info' : 'Concurrency info'}</Subheading>
+            </Box>
+            <Box padding={{bottom: 24}}>
+              <MetadataTableWIP style={{marginLeft: -1}}>
+                <tbody>
+                  {flagPoolUI ? (
+                    <tr>
+                      <td style={{verticalAlign: 'middle'}}>Granularity</td>
+                      <td>Op</td>
+                    </tr>
+                  ) : null}
+                  <tr>
+                    <td style={{verticalAlign: 'middle'}}>Limit</td>
+                    <td>
+                      <Box flex={{justifyContent: 'space-between', alignItems: 'center'}}>
+                        <div>
+                          {concurrencyLimit.slotCount ? (
+                            concurrencyLimit.slotCount
+                          ) : concurrencyLimit.limit === null ? (
+                            <>&mdash;</>
+                          ) : (
+                            concurrencyLimit.limit
+                          )}
+                        </div>
+                        <Button icon={<Icon name="edit" />} onClick={() => setShowEdit(true)}>
+                          Edit limit
+                        </Button>
+                      </Box>
+                    </td>
+                  </tr>
+                </tbody>
+              </MetadataTableWIP>
+            </Box>
+            <Box
+              padding={{vertical: 16, horizontal: 24}}
+              flex={{direction: 'row', alignItems: 'center', justifyContent: 'space-between'}}
+            >
+              <Subheading>In progress</Subheading>
+            </Box>
+            <Box style={{marginLeft: -1}}>
+              <PendingStepsTable keyInfo={concurrencyLimit} refresh={refetch} />
+            </Box>
+          </Box>
+        ) : (
+          <Box padding={{vertical: 64}}>
+            <Spinner purpose="section" />
+          </Box>
+        )}
+      </div>
+      <EditConcurrencyLimitDialog
+        concurrencyKey={concurrencyKey}
+        open={!!showEdit}
+        onClose={() => setShowEdit(false)}
+        onComplete={refetch}
+        minValue={data?.instance.minConcurrencyLimitValue || DEFAULT_MIN_VALUE}
+        maxValue={data?.instance.maxConcurrencyLimitValue || DEFAULT_MAX_VALUE}
+      />
+      <DeleteConcurrencyLimitDialog
+        concurrencyKey={concurrencyKey}
+        open={!!showDelete}
+        onClose={() => setShowDelete(false)}
+        onComplete={onDelete}
+      />
+    </>
+  );
+};
+
+const isValidLimit = (
+  concurrencyLimit?: string,
+  minLimitValue: number = DEFAULT_MIN_VALUE,
+  maxLimitValue: number = DEFAULT_MAX_VALUE,
+) => {
+  if (!concurrencyLimit) {
+    return false;
+  }
+  const value = parseInt(concurrencyLimit);
+  if (isNaN(value)) {
+    return false;
+  }
+  if (String(value) !== concurrencyLimit.trim()) {
+    return false;
+  }
```

Comment:
What's the specific case here?

Reply:
I think if you tried to do some sort of decimal:

```
const valueStr = "1.3";
const value = parseInt(valueStr);  // evaluates to 1
```

Reply:
I think if you use `type="number"` on the input, it won't allow a decimal to be entered since the default stepping value is `1` and input is therefore restricted to integers.